napari/_vispy/camera.py

```diff
@@ -133,15 +138,37 @@ def perspective(self, perspective):
         self._3D_camera.fov = perspective
         self._view.camera.view_changed()
 
+    @property
+    def mouse_zoom(self):
+        return self._view.camera.mouse_zoom
+
+    @mouse_zoom.setter
+    def mouse_zoom(self, mouse_zoom):
+        self._view.camera.mouse_zoom = mouse_zoom
+
+    @property
+    def mouse_pan(self):
+        return self._view.camera.mouse_pan
+
+    @mouse_pan.setter
+    def mouse_pan(self, mouse_pan):
+        self._view.camera.mouse_pan = mouse_pan
```

Comment:
I sticked to the style that is used in this file, it doesn't have type annotations anywhere. Mixing different code styles in a single file is a bad practice - you should either refactor it completely or leave it as it is. Refactoring the code goes outside the scope of this PR.

Reply:
Having or not having type annotation is not the case of code style. In the past, we made mistakes and did not add type annotation when creating this file. At this moment, there are no free resources to add type annotation to all files. A lack of type annotation is no argument not to add it to a new code part if fixing the annotation of the rest of the file is not required for this (for example use of non-typed API). This is not the case with this PR.

Reply:
[comment above was edited for better clarification]

Reply:
@ksofiyuk to reiterate @Czaki's point, napari is a reasonably old code base, and type annotations were not nearly as popular when we started as they are now. So we've just been adding annotations whenever we write new code. Having said that I don't think this issue is blocking in this PR — let's focus on fixing the WarningEmitter + tests. If you want to commit the simple type annotations suggested by @Czaki it would be appreciated — even if it's icky mixing annotated/unannotated code! Note that it was a key design decision in Python typing to support this kind of progressive typing precisely for this reason. 😊